crates/rbuilder/src/mev_boost/mod.rs

```diff
@@ -225,6 +225,46 @@ pub enum RelayError {
     ConnectionError,
     #[error("Internal Error")]
     InternalError,
+    /// Patch to allow error redacting propagating a RelayError.
+    /// Sadly it's not possible to regenerate the reqwest::Error to remove the source.
+    #[error("Redacted request error {0}")]
+    RedactedRequestError(RedactedRequestErrorKind),
+}
+
+/// List of things we can check from a reqwest::Error with the available getters.
+#[derive(Debug)]
+pub enum RedactedRequestErrorKind {
+    Builder,
+    Request,
+    Redirect,
+    Status,
+    Body,
+    Decode,
+    Unknown,
+}
+
+impl std::fmt::Display for RedactedRequestErrorKind {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
```

Comment:
If this is a common annoyance we can use `strum` https://docs.rs/strum/latest/strum/derive.Display.html